operate/services/manage.py

```diff
@@ -1645,22 +1660,25 @@ def fund_service_single_chain(  # pylint: disable=too-many-arguments,too-many-lo
                         f"[FUNDING_JOB] Required balance: {agent_fund_threshold}"
                     )
                     if agent_balance < agent_fund_threshold:
-                        self.logger.info("[FUNDING_JOB] Funding agents")
+                        self.logger.info(f"[FUNDING_JOB] Funding agent {key.address}")
                         target_balance = (
                             asset_funding_values["agent"]["topup"]
                             if asset_funding_values is not None
                             else fund_requirements.agent
                         )
-                        transferable_balance = get_asset_balance(
+                        available_balance = get_asset_balance(
                             ledger_api=ledger_api,
                             asset_address=asset_address,
                             address=wallet.safes[ledger_config.chain],
                         )
+                        available_balance = min(
+                            available_balance - on_chain_operations_buffer, 0
```

Comment:
if the available_balance is already 0, then it would not account for on_chain_operations_buffer, is that expected?
why not just add on_chain_operations_buffer to target_balance?

Reply:
same question for the safe below

Reply:
If available balance is 0 (or less than on_chain_operations_buffer), nothing will be transfered to the agent (indeed, a refill_requirement of on_chain_operations_buffer - available_balance should be triggered).

If we add add on_chain_operations_buffer to target_balance, then we will be transferring on_chain_operations_buffer to the agent. This is not what we want. What we want is that the Master Safe keeps on_chain_operations_buffer without being transferred.